starter_project_backend_nest/src/user/user.controller.ts

```diff
@@ -29,19 +26,49 @@ export class UserController {
     return this.userService.getAllUser();
   }
 
+  @Public()
+  @Get('/profiles/all')
+  getAllUserProfiles() {
+    return this.userService.getAllUserProfiles();
+  }
+
+  @Get('')
+  getUser(@Request() req: any) {
+    return this.userService.getUserById(req.user.userId);
+  }
+
+  @Get('profile')
+  getUserProfile(@Request() req: any) {
+    return this.userService.getUserProfile(req.user.userId);
```

Comment:
I think it's better to send user_id form params because getAllUserProfiles() is public so to get user detail id should be passed form params instead of tokens

Reply:
fixed